### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by esc**
Here is some stuff to disentangle the convoluted logic present in this program.

### CODE REVIEW

**Comment by schlomo** [comment]
Maybe this is a really good moment to replace `/data/home` with a more
standard-compliant `/home`? This will also help a lot with SELinux
compatibility. I also don't see any reason for non-standard paths here.

**Comment by esc** [comment]
Strongly agree.

On 11 May 2016 17:01:21 CEST, Schlomo Schapiro notifications@github.com wrote:

> Maybe this is a really good moment to replace `/data/home` with a more
> standard-compliant `/home`? This will also help a lot with SELinux
> compatibility. I also don't see any reason for non-standard paths here.
> 
> On 11 May 2016 at 16:53, Valentin Haenel notifications@github.com
> wrote:
> 
> > Here is some stuff to disentangle the convoluted logic present in
> > this
> > 
> > ## program.
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> >   https://github.com/ImmobilienScout24/c-bastion/pull/17
> > Commit Summary
> > - we only need to create the /data/home once at the beginning
> > - simplify tests for check_and_add
> > - rename variable for readability
> > - make the exception inherit from Exception instead of
> >   BaseException
> > - simplify checking the valididity of the username
> > - tests for simpler checking of the username
> > - simplify user creation
> > - simplify program flow
> > - remove redundent comment
> > - move the deletion stuff to the end
> > 
> > File Changes
> > - _M_ src/main/python/c_bastion/index.py
> > 
> > https://github.com/ImmobilienScout24/c-bastion/pull/17/files#diff-0
> >    (106)
> > - _M_ src/unittest/python/index_tests.py
> > 
> > https://github.com/ImmobilienScout24/c-bastion/pull/17/files#diff-1
> >    (92)
> > 
> > Patch Links:
> > - https://github.com/ImmobilienScout24/c-bastion/pull/17.patch
> > - https://github.com/ImmobilienScout24/c-bastion/pull/17.diff
> > 
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly or view it on GitHub
> > https://github.com/ImmobilienScout24/c-bastion/pull/17
> 
> ---
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/ImmobilienScout24/c-bastion/pull/17#issuecomment-218486689

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by esc** [comment]
@schlomo I fixed the path.

**Comment by esc** [comment]
Could someone please merge this? I think the first round of refactor is done now.

File: src/main/python/c_bastion/index.py [hunk 1]
```diff
@@ -85,42 +70,20 @@ def username_exists(username):
...
 
 def useradd(username):
-    sh.useradd(username, '-b', PATH_PREFIX, '-p', '*', '-s', '/bin/bash')
+    sh.useradd(username, '-b', HOME_PATH_PREFIX, '-p', '*', '-s', '/bin/bash')
 
 
-def check_and_add(username):
-    """
-    Check if the user already exists.
+def check_and_create_homes():
```
Comment (nmohaupt) [inline src/main/python/c_bastion/index.py]: Da jetzt das Standard-Home verwendet wird, ist das pruefen und selber erzeugen nicht mehr notwendig?

**Comment by esc** [comment]
Thanks for merge.

**Pull request merged**